lib/chart-block-macro/extension.rb

```diff
@@ -14,17 +13,9 @@ class ChartBlockMacro < Extensions::BlockMacroProcessor
   named :chart
 
   def process(parent, target, attrs)
+    engine = (attrs.key? 'engine') ? attrs['engine'].downcase : 'c3js'
     raw_data = PlainRubyCSV.read(File.join parent.document.base_dir, attrs['data-uri'])
```

Comment:
Could you change this so that the data is read from the target and the type is the first positional attribute? Perhaps you should also follow the block extension by making engine the second positional attribute.

```
...
name_positional_attributes 'type', 'engine'
...
```

and

```
...
raw_data = PlainRubyCSV.read(File.join parent.document.base_dir, target)
html = ChartBackend.process engine, attrs['type'], raw_data
...
```

However, to resolve the target, you should use the file resolution built in to Asciidoctor core.

```
data_path = parent.normalize_asset_path(target, 'target')
raw_data = parent.read_asset(data_path, warn_on_failure: true, normalize: true)
```

We probably need a cleaner API in core for handling this because it still doesn't account for reading data from URIs. Perhaps file an issue to get this cleaned up with this extension as a reference.